### PR TITLE
Initial propososal

### PR DESCRIPTION
**Pull request by Truba**

### CODE REVIEW

File: Base VIPER Interfaces/BaseWireframe.swift [hunk 1]
```diff
@@ -1,46 +1,58 @@
...
 
-    unowned var navigationController: UINavigationController
+    private unowned var _viewController: UIViewController
+    private var vcDealloc: UIViewController?
 
-    init(navigationController: UINavigationController) {
-        self.navigationController = navigationController
+    init(viewController: UIViewController) {
+        _viewController = viewController
+        vcDealloc = viewController
```
Comment (fgulan) [inline Base VIPER Interfaces/BaseWireframe.swift]: Better naming like: `_storedViewController` and add some comment why it is needed (to retain view controller reference upon first access)

File: Base VIPER Interfaces/BaseWireframe.swift [hunk 2]
```diff
@@ -1,46 +1,58 @@
...
-            fromViewController.present(navigationController, animated: animated, completion: nil)
-        case .root:
-            navigationController.setViewControllers([viewController], animated: animated)
-        }
+}
+
+extension BaseWireframe {
+    
+    var viewController: UIViewController {
+        defer { vcDealloc = nil }
```
Comment (fgulan) [inline Base VIPER Interfaces/BaseWireframe.swift]: Maybe add some comment here also.

File: Base VIPER Interfaces/BaseWireframe.swift (hunk 1 above)
Comment (Adis) [reply Base VIPER Interfaces/BaseWireframe.swift]: Agree on this one, vcDealloc is a really shitty name :D

**Comment by vburojevic** [comment]
@idikic @damjanvujaklija your thoughts on this?

yay or nay for this change?

**Pull request merged**